src/components/PlayerPanel.tsx

```diff
@@ -46,6 +54,26 @@ function PlayerPanelComponent({ player, index, cols, rows, isEvenPlayerIndexNumb
 
   const [view, setView] = useState<PanelView>(PanelView.PANEL);
 
+  const [cardName, setCardName] = useState('');
+  const [cardImageUrl, setCardImageUrl] = useState<string | undefined>(undefined);
+
+  const handleCardSearch = async () => {
+    if (!cardName.trim()) return;
+
+    try {
+      const imageUrl = await fetchCardByName(cardName);
+      if (imageUrl) {
+        setCardImageUrl(imageUrl);
+      } else {
+        setCardImageUrl(undefined);
+        // optionally show a "not found" message here
```

Comment:
i told you to just send up what you had but would be nice to do something user facing here, otherwise were just swallowing the error essentially.